### PR TITLE
DOC: document release 1.14.5

### PR DESCRIPTION
**Pull request by mattip**
Create new bug-fix release documents and update RELEASE_WALKTHROUGH from master (with a frew changes)

### CODE REVIEW

File: doc/RELEASE_WALKTHROUGH.rst.txt
Comment (rgommers) [inline doc/RELEASE_WALKTHROUGH.rst.txt]: This is not quite correct. Only important fixes need backporting; things that can wait should wait. There's some judgement involved here, but e.g. backporting doc typo fixes should not be done.

File: doc/RELEASE_WALKTHROUGH.rst.txt
Comment (rgommers) [inline doc/RELEASE_WALKTHROUGH.rst.txt]: not sure what these more strict requirements are, I think all policies are identical?

**Comment by rgommers** [comment]
LGTM modulo my two small comments.

File: doc/RELEASE_WALKTHROUGH.rst.txt
Comment (ahaldane) [reply doc/RELEASE_WALKTHROUGH.rst.txt]: No, numpy has a test [here](https://github.com/numpy/numpy/blob/master/numpy/tests/test_numpy_version.py) which limits what can go in the version number. See how I tried to make the version number "1.14-dev" in the second to last commit [here](https://github.com/numpy/numpy/commits/b8df705bdcce92d3e2c6f050eb4414192cf0df04), like I saw you do in scipy, but numpy didn't like it, so I had to push a followup commit.

That's why I wrote this here.

**Comment by ahaldane** [comment]
I presume you will forward port the changes in `doc/RELEASE_WALKTHROUGH.rst.txt` to master. It would help to see what you added relative to master.

File: doc/RELEASE_WALKTHROUGH.rst.txt
Comment (rgommers) [reply doc/RELEASE_WALKTHROUGH.rst.txt]: Ah that. That's the same for SciPy, there's just no test for it. `1.14-dev` is not a valid version (says PEP440), so I don't think SciPy has had such a thing recently.

I think the sentence just raises unnecessary questions, it's better to remove it. If you want to keep something here, I suggest a link to https://docs.scipy.org/doc/scipy/reference/dev/index.html#version-numbering or to copy that info over to the NumPy devguide.

Comment (rgommers) [reply doc/RELEASE_WALKTHROUGH.rst.txt]: And I should check before I write next time: scipy has the same test, https://github.com/scipy/scipy/blob/master/scipy/_build_utils/tests/test_scipy_version.py

Comment (ahaldane) [reply doc/RELEASE_WALKTHROUGH.rst.txt]: OK, that makes sense. I could have sworn I saw a REL commit in scipy that manually added `-dev`, but maybe I imagined it.

**Comment by charris** [comment]
Note that the current document lives in master, the one in maintenance is outdated and I haven't bothered to update it. It is easy to view the current version online here.

**Comment by mattip** [comment]
So the status of that document here is not any kind of blocker for the bugfix release, and should not postpone the merge of this branch into maintenance. If there are no other fixes that need to be in the bugfix, I will merge and continue the process.

**Comment by charris** [comment]
Why would you merge this into maintenance, it isn't needed and doesn't fix any bugs.

**Comment by mattip** [comment]
removed doc/RELEASE_WALKTHROUGH.rst.txt

**Comment by charris** [comment]
Just a reminder that you should check your cython version to make sure it is `0.28.2` for generating the source releases. I've also been renaming the one generated `README` file to `README.rst` to compliment `README.md`. That is fixed in the master `pavement.py` file, but not in maintenance.

**Review by charris — APPROVED** [review]

**Comment by charris** [comment]
You can merge this when you are ready, release managers have special, if temporary, privilege ;)

**Comment by mattip** [comment]
OK, I think I will go with the easy path and not add this PR

**Comment by mattip** [comment]
@charris The .travis.yml for the [current v1.14.x HEAD of numpy-wheels](https://github.com/MacPython/numpy-wheels/blob/v1.14.x/.travis.yml) lists cython 0.26.1, as does [appveyor.yml](https://github.com/MacPython/numpy-wheels/blob/v1.14.x/appveyor.yml#L28). Is that correct?

**Comment by charris** [comment]
It is OK for the wheels as there are no wheels for Python 3.7. But folks are building the source releases with 3.7 pre-releases and cythonizing those with the latest is needed for that.

**Comment by mattip** [comment]
@juliantaylor @pv, @rgommers In order to complete the release, we need to update the documentation on `docs.scipy.org`, I do not have permission to `make upload`. It seems 1.14.3, 1.14.4 also did not make the documentation site. In truth, this bug-fix release has no documentation changes, so perhaps we should skip that step anyway?

**Comment by rgommers** [comment]
I'll upload them.

**Pull request merged**
